### PR TITLE
Add testing

### PR DESCRIPTION
**Pull request by mahesh-panchal**
- Update .gitignore
- Move custom resources to site profiles (dardel, uppmax, nac)
- Add nf-test
- Update test profile to use test dataset
- Update pairtools process
- Add config README
- Remove old obsolete tests

### CODE REVIEW

File: tests/data/tiny/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Test data set provenance
+
+## Authors
+
+- Guilherme Dias
+- Martin Pippel
+
+## Provenance
```
Comment (mahesh-panchal) [inline tests/data/tiny/README.md:8]: This needs to be filled in

Comment (MartinPippel) [inline tests/data/tiny/README.md:8]:
```suggestion
## Provenance

- species: _Drosophila melanogaster_ 
- PacBio HiFi data was downloaded from NBCI´s SRA [SRR10238607](https://www.ncbi.nlm.nih.gov/sra/SRR10238607)
- Illumina HiC data was downloaded from NBCI´s SRA [SRR10512944](https://www.ncbi.nlm.nih.gov/sra/SRR10512944)

- PacBio HiFi data were randomly subsampled to 15X read coverage and assembled with [hifiasm version 0.19.8](https://github.com/chhylp123/hifiasm)
- PacBio reads were mapped back to the genome assembly with minimap2
- Illumina HiC reads were mapped back to the genome assembly with bwa mem

- PacBio and HiC reads were extracted from the alignment bam files: 
   - from a 2Mb region within a 28Mb contig
   - a 20Kb gap was introduces at Position 1Mb and all PacBio and HiC reads that mapped into that region were filtered out
   - HiC reads were further filtered down to 50%, representing a 38X coverage 
```

**Review by MartinPippel — APPROVED** [review]
- looks good to me 
- I like the idea to separate the resources requirements into another file tool_resources.config

File: tests/data/tiny/README.md (hunk 1 above)
Comment (mahesh-panchal) [reply tests/data/tiny/README.md]: Could you add code here too please? Did you use `wget` to fetch the reads.

**Pull request merged**